### PR TITLE
Fix date/datetime segment filter validation

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This is regression https://github.com/mautic/mautic/pull/12572

Some of input of dates was not validated properly

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create a contact date time field
3. Try create segment with filter value for the field `2023-07-07 00:00`
4. Try create segment with filter value for the field `birthday`
5. Both failed
6. After PR should passed

### CODE REVIEW

**Comment by RCheesley** [comment]
Looks like some tests need attention @kuzmany

**Review by PatrickJenkner — APPROVED** [review]
Worked as described for a user.

**Comment by PatrickJenkner** [comment]
> Worked as described for a user.

Using 'birthday' no longer works for me as of today, it gets saved as '1970-01-01 00:00' instead.
Other strings like 'today' work though

Edited for details, thanks @MadlenF

**Review by MadlenF — CHANGES_REQUESTED** [review]
Once I type in "birthday" and save the changes, they are saved as "1970-01-01 00:00"
Bildschirm­foto 2023-08-18 um 15 13 17

**Comment by RCheesley** [comment]
@kuzmany could you take a look at the feedback, seems something is not working as it should. Thanks!

**Comment by kuzmany** [comment]
@MadlenF I assume you are not using English language in your profile/configuration? But this issue cover another PR https://github.com/mautic/mautic/pull/12411

**Comment by MadlenF** [comment]
> @MadlenF I assume you are not using English language in your profile/configuration? But this issue cover another PR #12411

@kuzmany I opened the PR with Gitpod and haven't changed the language, so it's in English (US). However, behavior is as described as above.

**Comment by kuzmany** [comment]
@MadlenF you're testing it on the DateTime field. However, birthday supported just the date type  field.

**Comment by MadlenF** [comment]
Oh yes, that's what I understood from the testing instructions. For date field it's working 👍

File: app/bundles/LeadBundle/Validator/Constraints/SegmentDateValidator.php [hunk 1]
```diff
@@ -28,11 +28,20 @@ public function validate($filters, Constraint $constraint): void
                     continue;
                 }
 
-                $dateString = str_replace('%', '', $parameterValue);
+                if (str_contains($parameterValue, '%')) {
+                    return;
+                }
```
Comment (escopecz) [inline app/bundles/LeadBundle/Validator/Constraints/SegmentDateValidator.php:42]: Why was this changed? Can you add some value containing the `%` character to your test? Perhaps I don't get why was it replaced with an empty string in the previous version of the code.

Comment (kuzmany) [reply app/bundles/LeadBundle/Validator/Constraints/SegmentDateValidator.php:42]: For contains/startwith expressions string ouput value is like `%2023-01-01`. 
These formats I cannot validate with \DateTime::createFromFormat

But anyway, I marked it as WIP still need some things to do, hope tomorrow brings an update

This should validate potential issues with typos etc..  Often happen for our customers

Comment (kuzmany) [reply app/bundles/LeadBundle/Validator/Constraints/SegmentDateValidator.php:42]: I've added more commits.
We tried to validate some mistakes. 
For example, empty/not empty or regexp we have to skip. Also values for like not like expressions are skipped.
Now should cover all possible values to validate

**Comment by RCheesley** [comment]
This one needs re-testing after the code review changes if you have time folks!

**Review by PatrickJenkner — APPROVED** [review]
Still works as described 👍

**Review by escopecz — APPROVED** [review]
The code looks good, thanks Zdeno! 👍

**Comment by vinyll** [comment]
Hey there,
I'm new to Mautic and I need to play with segmentation.
Would you have any clue how to automate sending an email to users where birthday is in 5 days (to send them a gift email)?
Thanks!

**Comment by escopecz** [comment]
Hey @vinyll! Please use the Mautic forum for these types of questions. But in short, you can use relative date in the segment filter. Try this value for the birthday field:

`-5 days`
